cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go

```diff
@@ -378,29 +387,52 @@ func TestNodeGroupIncreaseSize(t *testing.T) {
 	t.Run("MachineSet", func(t *testing.T) {
 		tc := testCase{
 			description: "increase by 1",
-			initial:     3,
+			initialSpec: pointer.Int32Ptr(3),
 			expected:    4,
 			delta:       1,
 		}
-		test(t, &tc, createMachineSetTestConfig(testNamespace, int(tc.initial), annotations))
+		test(t, &tc, createMachineSetTestConfig(testNamespace, int(*tc.initialSpec), annotations))
 	})
 
 	t.Run("MachineDeployment", func(t *testing.T) {
 		tc := testCase{
 			description: "increase by 1",
-			initial:     3,
+			initialSpec: pointer.Int32Ptr(3),
 			expected:    4,
 			delta:       1,
 		}
-		test(t, &tc, createMachineDeploymentTestConfig(testNamespace, int(tc.initial), annotations))
+		test(t, &tc, createMachineDeploymentTestConfig(testNamespace, int(*tc.initialSpec), annotations))
+	})
+
```

Comment:
just a question, since we are adding more tests here should this function be refactored to use the looping style of test run?

Reply:
Could do, but in this case I don't think it makes much of a difference. The benefit of doing that is to share a single test function with some inputs, which we are already doing. I can change it if you'd prefer though?

Reply:
no, i don't think it's necessary to create another patchset. i was mainly curious to hear your thoughts about it since it's a pattern we seem to use in other places.